src/shared/seccomp-util.c

```diff
@@ -1452,7 +1451,13 @@ int seccomp_restrict_archs(Set *archs) {
         if (r < 0)
                 return r;
 
-        return seccomp_load(seccomp);
+        r = seccomp_load(seccomp);
+        if (IN_SET(r, -EPERM, -EACCES))
+                return r;
+        if (r < 0)
+                log_debug_errno(r, "Failed to restrict system call architectures, skipping: %m");
+
```

Comment:
I am bit concerned about ignoring errors here, if users tried to explicitly set it then and we ignore other architectures then those calls can be used to bypass the native architecture. Maybe turn these debug messages into warnings ? at least users can act on that ?

Reply:
yeah, it sucks, but there's little we can do until libseccomp reports proper errors back we can match against. There are cases we validly want to ignore errors and cases where we shouldn't. right now we propagate only EPERM/EACCESS as we know under which conditions those are generated. We should invert this all of course, but we can't as long as we don't have any error codes to check for.

Specifically, we want to at least ignore issues regarding:
1. syscall doesn't exist on some arch
2. syscall cannot matched on some arch (because multiplexed, think socket/socketcall)

and there are some more, but as long as we can't detect these properly we don't really have any realistic option than to ignore most, but propagate only the ones we know aren't OK.

Reply:
btw, do note that this PR doesn't really change much in this regard, in just adjusts one utlier to match what all the other invocations of seccomp_load() do here. Reworking the error handling of our seccomp_load() calls certainly should be revisited but probably later on, and then for all cases at once.